client/clients/router/client.go

```diff
@@ -446,8 +507,25 @@ batchLoop:
 				continue batchLoop
 			default:
 			}
+			// Check whether allow the follower to handle this batch of requests.
+			allowFollowerHandle := c.option.GetEnableFollowerHandle()
+			if allowFollowerHandle {
+				// We need to ensure all requests in a same batch allow to be handled by the follower.
+				c.batchController.IterCollectedRequests(func(req *Request) bool {
+					if !req.options.AllowFollowerHandle {
```

Comment:
What if the AllowFollowerHandle changed when IterCollectedRequests?

Reply:
I added one last check before choosing the stream, PTAL.